assets/js/swedbank-pay-design-guide-theme.js

```diff
@@ -21,51 +21,45 @@
+    var tocLinks = document.querySelectorAll("nav.sidebar-nav .nav-subgroup.active .nav-leaf");
+
+    if (tocLinks.length == 0) {
+        var tocLinks = document.querySelectorAll("nav.sidebar-nav .doc-toc .nav-leaf");
+    }
 
     var getPosition = function (el) {
         if (el) {
             var bodyRect = document.body.getBoundingClientRect();
             var elemRect = el.getBoundingClientRect();
-
+            
             return elemRect.top - bodyRect.top;
         }
-
+        
         return null;
     };
-
+    
     window.addEventListener("scroll", function () {
-        var buffer = document.body.clientHeight * 0.1;
-        var currentPos = window.pageYOffset + buffer;
-
-        // TODO: Probably a stupid way to compute "how far left can we scroll until
-        //       we reach the bottom of the page", but it seems to work.
-        var scrollDistanceFromBottom = document.documentElement.scrollHeight
-            - document.documentElement.scrollTop
-            - document.body.clientHeight
-            - buffer;
-
-        if (scrollDistanceFromBottom <= 0) {
-            for (var link of tocLinks) {
-                link.parentElement.classList.remove("active");
-            }
-
-            tocLinks[tocLinks.length - 1].parentElement.classList.add("active");
-            return;
-        }
-
-        for (var i = 0; i < headings.length; i++) {
-            var heading = headings[i];
-            var headingPos = getPosition(heading);
-            var nextHeadingPos = getPosition(headings[i + 1]);
-
-            if (currentPos > headingPos && currentPos < nextHeadingPos) {
-                for (var link of tocLinks) {
-                    link.parentElement.classList.remove("active");
-                }
+        if (tocLinks.length > 0) {
+            var activeLeaf = document.querySelector("nav.sidebar-nav .nav-leaf.active")
+            var buffer = document.body.clientHeight * 0.1;
+            var currentPos = window.pageYOffset + buffer;
+    
+            // TODO: Probably a stupid way to compute "how far left can we scroll until
+            //       we reach the bottom of the page", but it seems to work.
+            var scrollDistanceFromBottom = document.documentElement.scrollHeight
+                - document.documentElement.scrollTop
+                - document.body.clientHeight
+                - buffer;
+                
+            const scrollNumber = [...headings].filter(heading => getPosition(heading) <= currentPos).length - 1;
 
-                tocLinks[i].parentElement.classList.add("active");
-                return;
+            activeLeaf && activeLeaf.classList.remove("active")
+            
+            
+            if (scrollDistanceFromBottom > 0) {
+                scrollNumber >= 0 && tocLinks[scrollNumber].classList.add("active");
+            } else {
+                tocLinks[tocLinks.length - 1].classList.add("active");

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['tocLinks' used outside of binding context.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=5659803)